### PR TITLE
fix: catalyst cardano support for NFTs

### PR DESCRIPTION
**Pull request by dtscalac**
# Description

The PR adds support for NFTs in UTXOs so that we can send NFTs from one wallet to another. NFT support is required even if we don't intend to transfer NFTs since Cardano blockchain requirement is to spent all amounts in UTXOs, including NFTs. 

Algorithms taken from [Cardano multiplatform lib](https://github.com/dcSpark/cardano-multiplatform-lib/blob/develop/chain/rust/src/builders/tx_builder.rs#L1553)

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices_packages/catalyst_cardano_serialization/lib/src/types.dart [hunk 1]
```diff
@@ -70,3 +73,240 @@ extension type const SlotBigNum(int value) {
   /// Serializes the type as cbor.
   CborValue toCbor() => CborSmallInt(value);
 }
+
+/// Represents the balance of the wallet in terms of [Coin].
+///
+/// For now, other assets than Ada are not supported and are ignored.
+final class Value extends Equatable {
```
Comment (minikin) [inline catalyst_voices_packages/catalyst_cardano_serialization/lib/src/types.dart:80]: Can we think of a better name than `Value`? Value is too generic and can be misleading.
Based on the comments to the class probably we can use `Balance`.

**Review by minikin — CHANGES_REQUESTED** [review]

File: catalyst_voices_packages/catalyst_cardano_serialization/lib/src/types.dart (hunk 1 above)
Comment (dtscalac) [reply catalyst_voices_packages/catalyst_cardano_serialization/lib/src/types.dart]: The name comes from official CIP-30 docs but I agree, it's too vague, renamed to Balance.

Comment (minikin) [reply catalyst_voices_packages/catalyst_cardano_serialization/lib/src/types.dart]: We don't need to follow bad code practices ;)

**Review by minikin — APPROVED** [review]
lgtm

**Pull request merged**